src/core/memory.cc

```diff
@@ -1149,7 +1149,7 @@ void cpu_pages::do_resize(size_t new_size, allocate_system_memory_fn alloc_sys_m
     mem.release();
     maybe_enable_transparent_hugepages(mmap_start, mmap_size);
     // one past last page structure is a sentinel
-    auto new_page_array_pages = align_up(sizeof(page[new_pages + 1]), page_size) / page_size;
+    auto new_page_array_pages = align_up(sizeof(page) * (new_pages + 1), page_size) / page_size;
```

Comment:
Good fix. This wasn't even a VLA - no VLA was ever created here. It was just a weird way to do multiplication - it's like doing `sizeof(int[n])` instead of doing `sizeof(int)*n`. Good riddance.